frontend/packages/kubevirt-plugin/src/components/modals/add-template-source/add-template-source.tsx

```diff
@@ -0,0 +1,518 @@
+    kind: StorageClassModel.kind,
+    isList: true,
+    namespaced: false,
+  });
+
+  const defaultSCName = getDefaultStorageClass(storageClasses)?.metadata.name;
+
+  const handleStorageClass = (scName: string) => {
+    const updatedStorageClass = storageClasses.find((sc) => sc.metadata.name === scName);
+    const provisioner = updatedStorageClass?.provisioner || '';
+    const modes =
+      provisionerAccessModeMapping[provisioner] || getAccessModeForProvisioner(provisioner);
+    setAccessMode('ReadWriteOnce');
+    setAccessModes(modes);
+    setStorageClass(updatedStorageClass?.metadata?.name);
+  };
+
+  if (scLoaded && !storageClass) {
+    handleStorageClass(defaultSCName);
+  }
+
+  const onSubmit = async () => {
+    setAllocating(true);
+    setSubmitting(true);
+    const dvObj = getDataVolume(
+      { name: baseImageName, namespace: baseImageNamespace },
+      { ...size, accessMode, storageClass },
+      { dataSource, url, pvcName, pvcNamespace, pvcSize, container },
+    );
+    try {
+      if (dataSource === DATA_SOURCE.UPLOAD) {
+        const { token } = await createUploadPVC(dvObj);
+        setAllocating(false);
+        uploadData({
+          file: file.value,
+          token,
+          pvcName: dvObj.metadata.name,
+          namespace: dvObj.metadata.namespace,
+        });
+      } else {
+        await k8sCreate(DataVolumeModel, dvObj);
+      }
+      close();
+    } catch ({ message }) {
+      setError(message || 'Could not create persistent volume claim.');
+    } finally {
+      setAllocating(false);
+      setSubmitting(false);
+    }
+  };
+
+  let valid = scLoaded && !isSubmitting;
+  switch (dataSource) {
+    case DATA_SOURCE.UPLOAD: {
+      valid = valid && !!file?.value;
+      break;
+    }
+    case DATA_SOURCE.CLONE: {
+      valid = valid && !!pvcNamespace && !!pvcName;
+      break;
+    }
+    case DATA_SOURCE.URL: {
+      valid = valid && !!url;
+      break;
+    }
+    case DATA_SOURCE.REGISTRY: {
+      valid = valid && !!container;
+      break;
+    }
+    default:
+      valid = false;
+  }
+
+  const isUpstream = window.SERVER_FLAGS.branding === 'okd';
+
+  return (
+    <div className="modal-content modal-content--no-inner-scroll">
+      <ModalTitle>Add source to vendor template</ModalTitle>
+      {uploadAllowedLoading ? (
+        <LoadingBox />
+      ) : uploadAllowed ? (
+        <>
+          <ModalBody>
+            {!isSubmitting && (
+              <>
+                <div className="kubevirt-modal-desc">
+                  This data can be found in{' '}
+                  <b>
+                    Storage &gt; {PersistentVolumeClaimModel.labelPlural} &gt; {baseImageName}
+                  </b>{' '}
+                  under the <b>{baseImageNamespace}</b> namespace.
+                </div>
+
+                <Form>
+                  <FormRow fieldId="form-data-source" title="Data source type" isRequired>
+                    <Select
```

Comment:
We should use the new `<FormPFSelect />` instead of all new `<Select />`.
It handles the `isOpen` state for you

Reply:
that's handy. Fixed.